tools/RAiDER/statsPlot.py

```diff
@@ -598,22 +618,22 @@ def create_DF(self):
             append_poly.append(Polygon(np.column_stack((np.array([bbox[2], bbox[3], bbox[3], bbox[2], bbox[2]]),
                                                         np.array([bbox[0], bbox[0], bbox[1], bbox[1], bbox[0]])))))  # Pass lons/lats to create polygon
 
-        # check for grid cell intersection with each station (loop through each station).. fast, but assumes station locations don't change
+        # Check for grid cell intersection with each station
         idtogrid_dict = {}
-        unique_points = self.df.groupby(['ID', 'Lon', 'Lat']).size()
-        unique_points = [unique_points.index.get_level_values('ID').tolist(), unique_points.index.get_level_values(
-            'Lon').tolist(), unique_points.index.get_level_values('Lat').tolist()]
-        for i in unique_points[0]:
-            try:
-                coord = Point((unique_points[1][unique_points[0].index(
-                    i)], unique_points[2][unique_points[0].index(i)]))
-                idtogrid_dict[i] = [j.intersects(
-                    coord) for j in append_poly].index(True)
-            except:
-                idtogrid_dict[i] = 'NaN'
+        self.unique_points = self.df.groupby(['ID', 'Lon', 'Lat']).size()
+        self.unique_points = [self.unique_points.index.get_level_values('ID').tolist(), self.unique_points.index.get_level_values(
+            'Lon').tolist(), self.unique_points.index.get_level_values('Lat').tolist()]
+        # Initiate R-tree of gridded array domain
+        self.polygon_dict = dict((id(pt), i) for i, pt in enumerate(append_poly))
+        self.polygon_tree = STRtree(append_poly)
+        # Parallelize check of station intersection with grid cells
+        with multiprocessing.Pool(self.numCPUs) as multipool:
+            for stat_ID,grd_index in multipool.starmap(self._check_stationgrid_intersection, itertools.product(self.unique_points[0])):
```

Comment:
I would split the long line like this

```suggestion
            for stat_ID, grd_index in multipool.starmap(
                self._check_stationgrid_intersection, 
                itertools.product(self.unique_points[0])
            ):
```

Reply:
Ok, will move away from multiprocessing for this particular routine anyways though. Will give more detail in my update.